### PR TITLE
refactor WebserviceFileDataObject AuthModes

### PR DESCRIPTION
**Pull request by zzeekk**
### What changes are included in the pull request?
see #301

### CODE REVIEW

File: sdl-core/src/test/scala/io/smartdatalake/testutils/TestUtil.scala [hunk 1]
```diff
@@ -158,7 +158,7 @@ object TestUtil extends SmartDataLakeLogger {
           .port(port)
           .httpsPort(httpsPort)
           .bindAddress(host)
-          .keystorePath("src/test/resources/test_keystore.pkcs12")
+          .keystorePath("sdl-core/src/test/resources/test_keystore.pkcs12")
```
Comment (pgruetter) [inline sdl-core/src/test/scala/io/smartdatalake/testutils/TestUtil.scala:161]: @zzeekk I think this addition is the reason for the failed unit test:
`Cause: java.io.FileNotFoundException: /home/runner/work/smart-data-lake/smart-data-lake/sdl-core/sdl-core/src/test/resources/test_keystore.pkcs12 (No such file or directory)`

**Review by pgruetter — COMMENTED** [review]
Unit tests fail, please see comment

**Review by pgruetter — APPROVED** [review]
Nice, works correctly.
@imagar This refactoring will also be interesting for you as it needs a refactoring of the Atlas exporter

**Comment by zzeekk** [comment]
Also merged to branch develop

**Pull request merged**